### PR TITLE
Added CentralScene Command support

### PR DESCRIPTION
**Pull request by bluewalk**
Replacement for PR #15

### CODE REVIEW

File: ZWaveLib/Enums/Command.cs [hunk 1]
```diff
@@ -139,7 +139,12 @@ public enum Command : byte
         ClockVersion = 0x01,
         ClockGet = 0x05,
         ClockSet = 0x04,
-        ClockReport = 0x06,
+        ClockReport = 0x06,
+
+        // CentralScene
+        CentralSceneCapabilityGet = 0x01,
+        CentralSceneCapabilityReport = 0x02,
+        CentralSceneSet = 0x03,
```
Comment (Bounz) [inline ZWaveLib/Enums/Command.cs]: According to documentation this commands should be named
- CentralSceneSupportedGet
- CentralSceneSupportedReport
- CentralSceneNotification

respectively (see SDS13781-2 "Z-Wave Application Command Class Specification" paragraph 4.16 at http://zwavepublic.com/specifications).

File: ZWaveLib/Values/CentralSceneValue.cs [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+        Released = 0x1,
+        HeldDown = 0x2,
+        Pressed2times = 0x3,
+        Pressed3times = 0x04,
+        Pressed4times = 0x05,
+        Pressed5times = 0x06,
+        Reserved = 0x07,
+    }
+
+    public enum CentralSceneValueID
```
Comment (Bounz) [inline ZWaveLib/Values/CentralSceneValue.cs]: This enum is not used anywhere in the library.
Do we need it?

File: ZWaveLib/Values/CentralSceneValue.cs [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace ZWaveLib.Values
+{
+    public enum CentralScenePressType
+    {
+        Pressed1times = 0x0,
```
Comment (Bounz) [inline ZWaveLib/Values/CentralSceneValue.cs]: Please, rename to **Pressed1Time**

File: ZWaveLib/Values/CentralSceneValue.cs [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+        Released = 0x1,
+        HeldDown = 0x2,
+        Pressed2times = 0x3,
```
Comment (Bounz) [inline ZWaveLib/Values/CentralSceneValue.cs]: Please, rename this and following enum values to Pressed2**T**imes.
Also, please be consistent in using byte values (you are mixing 0x3 with 0x**0**3 style). Let it be 0x**0**3.

File: ZWaveLib/Values/CentralSceneValue.cs [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+        public byte Level; // byte 2 
+        public CentralScenePressType PressType; // byte 3
+
+        public static CentralSceneValue Parse(byte[] message)
+        {
+            int keyAttribute = message[3];
+
+            // Hack for Fibaro Keyfob
+            if (keyAttribute > 0x80)
+                keyAttribute -= 0x80;
```
Comment (Bounz) [inline ZWaveLib/Values/CentralSceneValue.cs]: This looks strange.
I think you have found that Fibaro Keyfob implements Central Scene Command Class V3, while you are implementing a mix of Central Scene Command Class V1 and V2.
In V3 there was introduced Slow Refresh flag in Central Scene Notification Command (see SDS13781-2 paragraph 4.17.7) which is the most significant bit, that's why you are subtracting the value of 0x80.
I think you should read Key Attribute from 3 least significant bits only.

File: ZWaveLib/Enums/EventParameter.cs [hunk 6]
```diff
@@ -78,7 +78,9 @@ public enum EventParameter
         SecurityGeneratedKey,
         DoorLockStatus,
         RoutingInfo,
-        Clock,
+        Clock,
+        CentralScene,
+        CentralSceneCount,
```
Comment (Bounz) [inline ZWaveLib/Enums/EventParameter.cs]: Don't you mind to rename this to CentralSceneNotification and CentralSceneSupportedReport?

File: ZWaveLib/CommandClasses/CentralScene.cs [hunk 7]
```diff
@@ -0,0 +1,72 @@
...
+
+        public static ZWaveMessage Get(ZWaveNode node)
+        {
+            return node.SendDataRequest(new byte[] {
+                (byte)CommandClass.CentralScene,
+                (byte)Command.CentralSceneCapabilityGet
+            });
+        }
+
+        public static ZWaveMessage Report(ZWaveNode node)
```
Comment (Bounz) [inline ZWaveLib/CommandClasses/CentralScene.cs]: I don't think that we are going to send Central Scene Notifications from the controller.

File: ZWaveLib/CommandClasses/CentralScene.cs [hunk 8]
```diff
@@ -0,0 +1,72 @@
...
+
+                case Command.CentralSceneCapabilityReport:
+                    nodeEvent = new NodeEvent(node, EventParameter.CentralSceneCount, (int)message[2], 0);
+                    break;
+            }
+
+            return nodeEvent;
+        }
+
+        public static ZWaveMessage Get(ZWaveNode node)
```
Comment (Bounz) [inline ZWaveLib/CommandClasses/CentralScene.cs]: It's better to rename this method to SupportedGet so it describes what this method actually do.

**Review by Bounz — CHANGES_REQUESTED** [review]

**Comment by bluewalk** [comment]
I have implemented the requested changes into the PR

**Review by Bounz — APPROVED** [review]

**Pull request merged**